Services/ReaderService.py

```diff
@@ -18,11 +20,29 @@ def get_instance():
             return ReaderService()
         return ReaderService.__instance
 
-    def read_transaction(self, block_file_path, transaction_hash):
-        # might be replaced by a hdfs command to read file
-        block_file_path = './BlockChain/' + block_file_path + ".json"
+    def read_from_local(self, block_hash):
+        block_file_path = self.writer_service.config['LOCAL_PATH'] + block_hash + ".json"
         with open(block_file_path, "r") as read_file:
             data = json.load(read_file)
+        return data
+
+
+    def read_from_hdfs(self, block_hash):
+        hdfs = HDFileSystem(host = self.writer_service.config['HDFS_HOST'], port = self.writer_service.config['HDFS_PORT'])
```

Comment:
Can the import `from hdfs3 import HDFileSystem` be moved in the `read_from_hdfs` function ? It will allow people without hadoop/hdfs3 also to run the application, without installing those beforehand. 
Make a try/except block here, where you try to import, and catch `ImportError` if the hdfs3 module is not available/installed.

Reply:
I have resolved it. Please check @digaru19